ddtrace/tracer/tracer.go

```diff
@@ -364,6 +397,9 @@ func newTracer(opts ...StartOption) *tracer {
 		t.reportHealthMetrics(statsInterval)
 	}()
 	t.stats.Start()
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, syscall.SIGTERM)
+	t.setupSignalListener(sigChan, 5*time.Second) // Default flush timeout of 5 seconds on sigterm
```

Comment:
What will happen for cases where the agent is not available ? 🤔 Will the app take exactly 5sec to close ? If this is the case I believe 3 or 2 seconds would be appreciated and enough to flush everything if the agent is available

Maybe making this duration a documented `const` would make sense 🤔